lib/react_client.dart

```diff
@@ -36,19 +36,100 @@ newJsMap(Map map) {
 }
 
 /**
- * Type of [children] must be child or list of childs, when child is JsObject or String
+ * Type of [children] must be child or list of children, when child is JsObject or String
  */
 typedef JsObject ReactComponentFactory(Map props, [dynamic children]);
 typedef Component ComponentFactory();
 
 class ReactComponentFactoryProxy implements Function {
-  final ReactComponentFactory _call;
   final JsFunction reactComponentFactory;
-  ReactComponentFactoryProxy(this.reactComponentFactory, this._call);
+  ReactComponentFactoryProxy(this.reactComponentFactory);
 
   JsObject call(Map props, [dynamic children]) {
-    return this._call(props, children);
+
+    if (children == null) {
+      children = [];
+    } else if (children is! Iterable) {
+      children = [children];
+    }
+
+    return reactComponentFactory.apply([
+      _generateExtendedJSProps(props, children),
+      new JsArray.from(children)
+    ]);
+  }
+
+  dynamic noSuchMethod(Invocation invocation) {
+    if ((invocation.memberName == #call) && (invocation.isMethod)) {
+      Map props = invocation.positionalArguments[0];
+      List<dynamic> children = invocation.positionalArguments.sublist(1);
+      List<dynamic> reactParams = [_generateExtendedJSProps(props, children)];
+      reactParams.addAll(children);
+      return reactComponentFactory.apply(reactParams);
+    }
+    return super.noSuchMethod(invocation);
+  }
+
+  _generateExtendedJSProps(Map props, List<dynamic> children) {
+
+    var extendedProps = new Map.from(props);
+    extendedProps['children'] = children;
+
+    var jsProps = newJsObjectEmpty();
+
+    /**
+     * add key to args which will be passed to javascript react component
+     */
+    if (extendedProps.containsKey('key')) {
+      jsProps['key'] = extendedProps['key'];
+    }
+
+    if (extendedProps.containsKey('ref')) {
+      jsProps['ref'] = extendedProps['ref'];
+    }
+
+    /**
+     * put props to internal part of args
+     */
+    jsProps[INTERNAL] = {PROPS: extendedProps};
+
+    return jsProps;
+  }
+
+}
+
+class ReactDomComponentFactoryProxy implements Function {
+  final String name;
+  ReactDomComponentFactoryProxy(this.name);
+
+  JsObject call(Map props, [dynamic children]) {
+    _convertProps(props);
+    if (children is Iterable) {
+      children = new JsArray.from(children);
+    }
+    return _React['createElement'].apply([name, newJsMap(props), children]);
+  }
+
+  dynamic noSuchMethod(Invocation invocation) {
+    if ((invocation.memberName == #call) && (invocation.isMethod)) {
+      Map props = invocation.positionalArguments[0];
+      _convertProps(props);
+      List<dynamic> reactParams = [name, newJsMap(props)];
+      List<dynamic> children = invocation.positionalArguments.sublist(1);
+      reactParams.addAll(children);
+      return _React['createElement'].apply(reactParams);
```

Comment:
Not sure exactly how it's handled under the hood, but [`JsObject.callMethod`](https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer/dart:js.JsObject#id_callMethod) might be preferable.
```dart
    return _React.callMethod('createElement', reactParams);
```